utils/python/CIME/XML/entry_id.py

```diff
@@ -140,6 +140,17 @@ def _get_description (self, node):
     def _get_group (self, node):
         return self._get_node_element_info(node, "group")
 
+    def get_valid_values(self, vid):
+        node = self.get_optional_node("entry", {"id":vid})
+        if node is None:
+            return None
+
+        vv_node = self.get_optional_node("valid_values", root=node)
```

Comment:
The code in lines 148-152 is pretty widely used.  I was about to tell you that I've added code in generic_xml.py that does this, but then I realised that that code is still on a branch...